apps/typegpu-docs/src/content/docs/guides/tgpu-cli.mdx

```diff
@@ -0,0 +1,521 @@
+      - <FileDiff> sprite.ts </FileDiff>
+</FileTree>
+</div>
+
+The generated TypeScript definitions look like this:
+
+<Tabs>
+  <TabItem label="updateSprites.ts">
+  <details>
+  <summary>Click to see the content</summary>
+  ```typescript
+  /* generated via tgpu-cli by TypeGPU */
+
+  import tgpu from 'typegpu';
+  import * as d from 'typegpu/data';
+
+  /* structs */
+  export const Particle = d.struct({
+    pos: d.vec2f,
+    vel: d.vec2f,
+  });
+
+  export const SimParams = d.struct({
+    deltaT: d.f32,
+    rule1Distance: d.f32,
+    rule2Distance: d.f32,
+    rule3Distance: d.f32,
+    rule1Scale: d.f32,
+    rule2Scale: d.f32,
+    rule3Scale: d.f32,
+  });
+
+  export const Particles = (arrayLength: number) => d.struct({
+    particles: d.arrayOf(Particle, arrayLength),
+  });
+
+
+  /* bindGroupLayouts */
+  export const layout0 = tgpu.bindGroupLayout({
+    params: {
+      uniform: SimParams,
+    },
+    particlesA: {
+      storage: Particles,
+      access: 'readonly',
+    },
+    particlesB: {
+      storage: Particles,
+      access: 'mutable',
+    },
+  }).$forceIndex(0);
+  ```
+  </details>
+  </TabItem>
+  <TabItem label="sprite.ts">
+  <details>
+  <summary>Click to see the content</summary>
+  ```typescript
+  /* generated via tgpu-cli by TypeGPU */
+
+  import tgpu from 'typegpu';
+  import * as d from 'typegpu/data';
+
+  /* structs */
+  export const VertexOutput = d.struct({
+    position: d.vec4f,
+    color: d.vec4f,
+  });
+  ```
+  </details>
+  </TabItem>
+</Tabs>
+
+These definitions can now be used in our TypeScript code to interact with the shaders in a type-safe manner.
+If we wanted to tinker with the shaders, but still have the updated definitions, we could use the `--watch` flag to keep the definitions up-to-date.
+For specific usage details, read the Usage .
+
+## Usage
+
+This section will cover the most common use cases of the TypeGPU CLI and provide short examples.
+
+### Generate TypeGPU definitions for WGSL shader files
+
+To generate TypeGPU definitions from a single WGSL shader file, run:
+
+```bash
+tgpu-cli gen path/to/shader.wgsl
+```
+
+This will generate a TypeScript file with the TypeGPU definitions in the same directory as the shader.
+By default, the generated file will have the same name as the shader, with the `.ts` extension.
+
+<div class="relative">
+<FileTree>
+  - path/to
+      - shader.wgsl
+      - <FileDiff> shader.ts </FileDiff>
+</FileTree>
+</div>
+
+:::note
+Another valid way to run the command is by using quotes:
+```bash
+tgpu-cli gen "path/to/shader.wgsl"
+# or
+tgpu-cli gen 'path/to/shader.wgsl'
+```
+The quotes are not necessary if the path does not contain spaces. However, using them is a good practice to avoid potential issues.
+:::
+
+TypeGPU CLI supports glob patterns for specifying input paths.
+The following command will generate TypeGPU definitions for all WGSL files inside a given directory:
+
+```bash
+tgpu-cli gen "path/to/*.wgsl"
```

Comment:
I would add somewhere around here a note, advising users to provide the pattern in quotes, to avoid matching the pattern by the shell before passing it as argument. Or combine it with the previous note, but make it more obvious that when using * or **, the quotes are usually a must (I assume, though it's difficult to tell how every single shell works)